### PR TITLE
Better heuristics around application health

### PR DESCRIPTION
**Pull request by adborden**
This is the first part in [fixing the app health indicator](https://github.com/18F/cg-dashboard/issues/860).

- [ ] This depends on styles in https://github.com/18F/cg-style/pull/276

screenshot from 2017-01-04 15-31-38

### CODE REVIEW

File: static_src/util/health.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+export function appHealth(app) {
+  if (!app) {
+    throw new Error('`app` must be provided.');
+  }
+
+  if (app.state === appStates.started && app.running_instances === app.instances) {
+    return entityHealth.ok;
+  }
+
+  if (app.running_instances === 0) {
```
Comment (msecret) [inline static_src/util/health.js]: Are we sure theres no case where running instances would be more then 1 and app state would be stopped?

File: static_src/constants.js [hunk 2]
```diff
@@ -2,15 +2,32 @@
...
   started: 'STARTED',
   stopped: 'STOPPED',
-  crashed: 'CRASHED',
-  restarting: 'RESTARTING',
-  none: 'NONE',
   default: 'STOPPED'
 };
 
+// Descriptive states to represent entity health
+const entityHealth = keymirror({
```
Comment (msecret) [inline static_src/constants.js]: I like how we are separating them, it's easier to understand

File: static_src/constants.js [hunk 3]
```diff
@@ -2,15 +2,32 @@
 import keymirror from 'keymirror';
 
 const appStates = {
+  crashed: 'CRASHED',
+  down: 'DOWN',
+  flapping: 'FLAPPING',
```
Comment (msecret) [inline static_src/constants.js]: ???

Comment (adborden) [inline static_src/constants.js]: I got this from the CLI, but not sure under what conditions all of these might appear.

File: static_src/util/health.js (hunk 1 above)
Comment (adborden) [inline static_src/util/health.js]: As far as I can tell, no. I suspect there might be some corner cases to handle. For example, I have seen `running_instances === -1`, when the app is restarting/restaging.

**Pull request merged**